torch/nn/modules/module.py

```diff
@@ -23,7 +23,17 @@ def _addindent(s_, numSpaces):
     return s
 
 
-class Module(object):
+class ModuleMeta(type):
+    def __instancecheck__(self, instance):
+        try:
+            _original_cls = instance._original_cls
```

Comment:
Tried to use getattr(), but got NotImplementedError for some modules. Also, tried to use isinstance(instance, WeakScriptModuleProxy). Couldn't import it due to a recursive import. Any idea for this?

Reply:
`torch.jit` depends on `torch.nn` since `torch.jit.ScriptModule` is a subclass of `torch.nn.Module`. I think if we want to support `isinstance` it has to be done only on `WeakScriptModuleProxy` (without making any changes to the inheritance chain of `nn.Module`). Maybe a less intrusive solution is to subclass `ScriptMeta`, add the `__instancecheck__` method, and somehow install that on only the `WeakScriptModuleProxy` class

Reply:
Agree, but `isinstance()` does not work as that way. When we `isinstance(module, nn.Linear)`, python does: `type(nn.Linear).__instancecheck__(nn.Linear, module)`. That's why I added `ModuleMeta` to cover every module's `__instancecheck__`. We don't have an option to implementing `__instancecheck__` on `ScriptMeta`, refer to https://stackoverflow.com/questions/52168971/instancecheck-overwrite-shows-no-effect-what-am-i-doing-wrong for more information.
If we're not going this way, there's a method providing `torch.isinstance()` as I suggested.

Reply:
Or... can we inherit from both class: `SomeProxy(WeakScriptModuleProxy, original_cls)`, when creating WeakScriptModuleProxy instance?